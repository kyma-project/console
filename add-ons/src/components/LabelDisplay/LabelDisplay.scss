.labels-display {

    display: inline-block;
  
    ul {
      display: inline;
  
      li {
        display: inline-block;
        margin: 8px 4px;
      }
    }
    
    .caption--no-button {
      cursor: default;
      &:after {
        all: unset;
      }
    }
}
  