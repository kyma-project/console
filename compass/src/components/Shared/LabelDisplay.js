import React from 'react';
import PropTypes from 'prop-types';
import { Token } from "@kyma-project/react-components";

LabelDisplay.propTypes = {
  className: PropTypes.string,
  labels: PropTypes.object.isRequired,
};

export default function LabelDisplay(props) {
  return (
    <span className={props.className}>
      {Object.keys(props.labels).map(label => (
        <Token key={label} className="y-fd-token y-fd-token--no-button">
          {label}=[{props.labels[label].join(', ')}]
        </Token>
      ))}
    </span>
  );
}
