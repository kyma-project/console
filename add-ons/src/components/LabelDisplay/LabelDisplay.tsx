import React from 'react';
import { Token } from 'fundamental-react';
import './LabelDisplay.scss';

interface LabelDisplayProps {
  readonlyLabels?: { [key: string]: any };
  labels?: { [key: string]: any };
}

const LabelDisplay: React.FunctionComponent<LabelDisplayProps> = ({
  readonlyLabels,
  labels,
}) => (
  <section className="labels-display">
    <ul>
      {readonlyLabels &&
        Object.keys(readonlyLabels).map((key: string) => (
          <li key={key}>
            <Token className="caption-muted caption--no-button">{`${key}=${readonlyLabels[key]}`}</Token>
          </li>
        ))}
      {labels &&
        Object.keys(labels).map((key: string) => (
          <li key={key}>
            <Token className="caption-muted">{`${key}=${labels[key]}`}</Token>
          </li>
        ))}
    </ul>
  </section>
);

export default LabelDisplay;
