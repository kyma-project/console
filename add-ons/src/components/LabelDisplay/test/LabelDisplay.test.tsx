import React from 'react';
import LabelDisplay from './../LabelDisplay';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

const mockLabels = {
  a: 1,
  b: 2,
};

const mockReadonlyLabels = {
  aR: 'a',
  bR: 'b',
};

describe('LabelDisplay', () => {
  it('Renders with minimal props', () => {
    const component = mount(<LabelDisplay />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Renders labels', () => {
    const component = mount(
      <LabelDisplay readonlyLabels={mockReadonlyLabels} labels={mockLabels} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
