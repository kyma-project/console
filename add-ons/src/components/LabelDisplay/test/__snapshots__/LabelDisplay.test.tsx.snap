// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelDisplay Renders labels 1`] = `
<LabelDisplay
  labels={
    Object {
      "a": 1,
      "b": 2,
    }
  }
  readonlyLabels={
    Object {
      "aR": "a",
      "bR": "b",
    }
  }
>
  <section
    className="labels-display"
  >
    <ul>
      <li
        key="aR"
      >
        <Token
          className="caption-muted caption--no-button"
        >
          <span
            className="fd-token caption-muted caption--no-button"
            role="button"
          >
            aR=a
          </span>
        </Token>
      </li>
      <li
        key="bR"
      >
        <Token
          className="caption-muted caption--no-button"
        >
          <span
            className="fd-token caption-muted caption--no-button"
            role="button"
          >
            bR=b
          </span>
        </Token>
      </li>
      <li
        key="a"
      >
        <Token
          className="caption-muted"
        >
          <span
            className="fd-token caption-muted"
            role="button"
          >
            a=1
          </span>
        </Token>
      </li>
      <li
        key="b"
      >
        <Token
          className="caption-muted"
        >
          <span
            className="fd-token caption-muted"
            role="button"
          >
            b=2
          </span>
        </Token>
      </li>
    </ul>
  </section>
</LabelDisplay>
`;

exports[`LabelDisplay Renders with minimal props 1`] = `
<LabelDisplay>
  <section
    className="labels-display"
  >
    <ul />
  </section>
</LabelDisplay>
`;
